Core/Extensions/Foundation/Dictionary.swift

```
@@ -31,5 +31,30 @@ public extension Dictionary {
             updateValue(value, forKey: key)
         }
     }
-    
+
+    /**
+         Returns the value associated with the specified key,
+         or the specified default value if the key didn't have a value.
+     
+         Uses autoclosure so your default value won't be created if it's not needed.
+     */
+    public func value(forKey key: Key, or defaultValue: @autoclosure () -> Value) -> Value {
+        return self[key] ?? defaultValue()
+    }
+
+}
+
+public extension Dictionary where Value == Any {
+
+    /**
+         Returns the value associated with the specified key casted to T
+         (the type of the default value), or the specified default value
+         if the key didn't have a value or if it couldn't be casted to T.
+
+         Uses autoclosure so your default value won't be created if it's not needed.
+     */
+    public func castedValue<T>(forKey key: Key, or defaultValue: @autoclosure () -> T) -> T {
+       return self[key] as? T ?? defaultValue()
```

I agree that we shouldn't do this, the only case where I see it as useful is in cases where you know that something is wrong and you want to ignore it (as in that example when API is returning a bad type)... but I don't think we should give that functionality because basically is making it easy for the user to do something wrong and ignore it. If he or she wants to do something like this for some reason, I would prefer that the user is responsible for that and has to explain why it needs it instead of using a method form our own library by default and hiding possible errors